mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java

```diff
@@ -114,4 +138,75 @@ public Subscription sink(@NotNull final Observable<SinkOperation<Send>> spout) {
             .observeOn(Rx.compute())
             .subscribe(subscriber);
     }
+
+    @VisibleForTesting
+    static final class UserAgent {
+        private static final Joiner JOINER = Joiner.on(" ").skipNulls();
+        @NotNull
+        private final List<UserAgentEntry> entries;
+
+        @SafeVarargs
+        public UserAgent(@NotNull final Function<Class<? extends UserAgent>, UserAgentEntry>... entries) {
```

Comment:
I've relaxed the generic to `Class<?>`. The reason we can't refer to it in the static form is that the class is being used to locate resources on the classpath (specifically pom.properties and gradle properties) to that we can pass the version of the client information along to the master.